--- a/src/init/ssdk_plat.c
+++ b/src/init/ssdk_plat.c
@@ -278,7 +278,8 @@ static sw_error_t qca_mii_reg_convert(a_uint32_t dev_id, a_uint32_t *reg)
 	return SW_OK;
 }
 
-#if IS_ENABLED(CONFIG_MDIO_BITBANG)
+#if IS_ENABLED(CONFIG_MDIO_BITBANG) && \
+	(LINUX_VERSION_CODE < KERNEL_VERSION(6, 12, 0))
 sw_error_t qca_bitbang_mii_raw_read_try(struct mii_bus *bus,
 				     a_uint32_t reg, a_uint32_t *val)
 {
@@ -447,7 +448,8 @@ sw_error_t qca_mii_raw_read(struct mii_bus *bus, a_uint32_t reg, a_uint32_t *val)
 		return _qce2204_ahb_read(bus, addr, reg, val);
 	}
 
-#if IS_ENABLED(CONFIG_MDIO_BITBANG)
+#if IS_ENABLED(CONFIG_MDIO_BITBANG) && \
+	(LINUX_VERSION_CODE < KERNEL_VERSION(6, 12, 0))
 	sw_error_t rv = SW_OK;
 
 	rv = qca_bitbang_mii_raw_read_try(bus, reg, val);
@@ -474,7 +476,8 @@ sw_error_t qca_mii_raw_write(struct mii_bus *bus, a_uint32_t reg, a_uint32_t val)
 		return _qce2204_ahb_write(bus, addr, reg, val);
 	}
 
-#if IS_ENABLED(CONFIG_MDIO_BITBANG)
+#if IS_ENABLED(CONFIG_MDIO_BITBANG) && \
+	(LINUX_VERSION_CODE < KERNEL_VERSION(6, 12, 0))
 	sw_error_t rv = SW_OK;
 
 	rv = qca_bitbang_mii_raw_write_try(bus, reg, val);
@@ -502,7 +505,8 @@ sw_error_t qca_mii_raw_update(struct mii_bus *bus, a_uint32_t reg,
 		return _qce2204_ahb_update(bus, addr, reg, clear, set);
 	}
 
-#if IS_ENABLED(CONFIG_MDIO_BITBANG)
+#if IS_ENABLED(CONFIG_MDIO_BITBANG) && \
+	(LINUX_VERSION_CODE < KERNEL_VERSION(6, 12, 0))
 	sw_error_t rv = SW_OK;
 
 	rv = qca_bitbang_mii_raw_update_try(bus, reg, clear, set);
